targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/TARGET_MCU_K64F/watchdog.c

```diff
@@ -0,0 +1,129 @@
+#include "watchdog_api.h"
+
+#include "reset_reason_api.h"
+
+#include "fsl_wdog.h"
+
+// Platform specific watchdog definitions
+#define LPO_CLOCK_FREQUENCY 1000
+#define MAX_PRESCALER       8
+#define MAX_TIMEOUT         0xFFFFFFFF
+
+// Number of decrements in the timeout register per millisecond
+#define TICKS_PER_MS (LPO_CLOCK_FREQUENCY / 1000)
+// Maximum timeout that can be specified in milliseconds
+#define MAX_TIMEOUT_MS ((MAX_TIMEOUT / TICKS_PER_MS) * MAX_PRESCALER)
```

Comment:
On GCC_ARM this gets truncated to `0xfffffff8` (from `0x7fffffff8` which is a result of `0xffffffff * 0x8`). I suggest defining `MAX_TIMEOUT` as `0xFFFFFFFFULL` and possibly changing `MAX_TIMEOUT_MS` to `const uint64_t`. Anyway all the variables representing timeouts in milliseconds should be `uint64_t` type.

Or maybe it would be acceptable to assert that `MAX_TIMEOUT_MS` (without truncation) isn't greater than `UINT32_MAX` and adjusting `MAX_PRESCALER` and `MAX_TIMEOUT` to fit?

Reply:
Yeah, multiplying a maxed integer by something is probably a bad idea. I'm not sure it's necessary to support a ms timeout of more than 32 bits, which is around 49 days? though. I'll change it to something sensible